Ability to configure global options

robin-thoni opened the pull request:
Issue #11

#### Bugfixes

* Fixed error when input are not in a form (cf2b00f)
* Fixed date object error (7417a93)

#### Features

* Added service to configure settings globally
* Added demo

robin-thoni committed:
Added minimal demo; updated bower/npm devDependencies for the demo; updated README for the demo

robin-thoni committed:
Added .idea in gitignore for Jetbrains IDE

robin-thoni committed:
Fixed date filter sending a moment date instead of Date object

robin-thoni committed:
Fixed error when no form

robin-thoni committed:
Added mdpLocale service with default values

robin-thoni committed:
Added service default values if no value given in directive

robin-thoni committed:
Updated demo

robin-thoni committed:
Fixed time picker min/max time

robin-thoni committed:
Updated demo for time picker fix

robin-thoni committed:
Removed lamdba style function for more compatibility

robin-thoni committed:
Added build target for demo

robin-thoni committed:
Updated README

robin-thoni committed:
Updated dist files

robin-thoni committed:
Added demo-dist

robin-thoni committed:
Added release How-to in README

robin-thoni committed:
Updated version and CHANGELOG

dpoetzsch commented on src/components/mdpDatePicker/mdpDatePicker.js:
```diff
@@ -423,15 +427,15 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
 
                 ngModel.$validators.minDate = function(modelValue, viewValue) {
-                    return minDateValidator(viewValue, scope.dateFormat, scope.minDate);
+                    return minDateValidator(viewValue, scope.dateFormat, minDate);
```
I think, here you break the possibility to change the `minDate` after creation of the picker using the scope binding. I believe we have to do a `scope.minDate || $mdpLocale.date.minDate` here (and cleanup the constant definition above).

dpoetzsch commented on src/components/mdpDatePicker/mdpDatePicker.js:
```diff
@@ -423,15 +427,15 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
 
                 ngModel.$validators.maxDate = function(modelValue, viewValue) {
-                    return maxDateValidator(viewValue, scope.dateFormat, scope.maxDate);
+                    return maxDateValidator(viewValue, scope.dateFormat, maxDate);
```
same as with `minDate`

dpoetzsch commented on src/components/mdpDatePicker/mdpDatePicker.js:
```diff
@@ -423,15 +427,15 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
 
                 ngModel.$validators.filter = function(modelValue, viewValue) {
-                    return filterValidator(viewValue, scope.dateFormat, scope.dateFilter);
+                    return filterValidator(viewValue, scope.dateFormat, dateFilter);
```
same as with `minDate`

dpoetzsch commented on src/components/mdpDatePicker/mdpDatePicker.js:
```diff
@@ -484,9 +488,9 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
-                        dateFilter: scope.dateFilter,
+                        minDate: minDate,
+                        maxDate: maxDate,
+                        dateFilter: dateFilter,
```
same as with `minDate` above (for all three lines)

dpoetzsch commented on src/components/mdpTimePicker/mdpTimePicker.js:
```diff
@@ -250,12 +250,12 @@ module.provider("$mdpTimePicker", function() {
                     time: time,
                     autoSwitch: options.autoSwitch,
-                    ampm: options.ampm
+                    ampm: options.ampm == null ? $mdpLocale.time.ampm : options.ampm
```
I think we might go better with `angular.isDefined(options.ampm) ? options.ampm : $mdpLocale.time.ampm` here

dpoetzsch commented on src/components/mdpTimePicker/mdpTimePicker.js:
```diff
@@ -369,11 +372,11 @@ module.directive("mdpTimePicker", ["$mdpTimePicker", "$timeout", function($mdpTi
             
             ngModel.$validators.minTime = function(modelValue, viewValue) {
-                return minTimeValidator(viewValue, scope.timeFormat, scope.minTime);
+                return minTimeValidator(viewValue, scope.timeFormat, minTime);
```
same as with `minDate` in date picker

dpoetzsch commented on src/components/mdpTimePicker/mdpTimePicker.js:
```diff
@@ -369,11 +372,11 @@ module.directive("mdpTimePicker", ["$mdpTimePicker", "$timeout", function($mdpTi
 
             ngModel.$validators.maxTime = function(modelValue, viewValue) {
-                return maxTimeValidator(viewValue, scope.timeFormat, scope.maxTime);
+                return maxTimeValidator(viewValue, scope.timeFormat, maxTime);
```
same as with `minDate` in date picker

dpoetzsch reviewed (changes requested):
Lovely, thanks!

I added some remarks in the code. Only little issues though.

Furthermore, it looks like all the files in demo-dist are auto-generated? If yes, we should exclude them from the repository.

Also, at some point we should merge your demo with the original one (which might be broken at the moment)... But this is work for another PR.

robin-thoni committed:
Fixed options to be editable at runtime; used angular.isDefined instead of == null

robin-thoni committed:
Added customizable demo

robin-thoni committed:
Removed demo-dist

robin-thoni commented:
> I added some remarks in the code. Only little issues though.

I fixed them

> Furthermore, it looks like all the files in demo-dist are auto-generated? If yes, we should exclude them from the repository.

I was hesitant about that before making the PR, and I ended by keeping it. Now, it is ignored.

> Also, at some point we should merge your demo with the original one (which might be broken at the moment)... But this is work for another PR.

Yes :) I spotted a few other bugs, I'll probably fix them
